dot/parachain/availability-store/messages.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package availability_store
```

Comment:
These types don't have to be varying data types. 

We need to make a type varying data type when we want to scale encode or decode it, which happens when 1) we use that type in a message sent to other node or 2) we use that type as an argument in a runtime call.

That is not the case here. The way to confirm if a type gets scale encoded or decoded is by checking the `#[derive]` attribute. If a type needs to be encoded or decoded attribute should be `#[derive(Encode, Decode)]`. 
If you look at this type, Encode and Decode traits are missing.
https://github.com/paritytech/polkadot-sdk/blob/cb944dc54882222453071fc99f2e5014c51823ef/polkadot/node/subsystem-types/src/messages.rs#L477